### PR TITLE
Connect authentication service to notification service

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Review by jusoto — APPROVED** [review]

File: build.gradle [hunk 1]
```diff
@@ -40,9 +40,10 @@ ext {
 
 dependencies {
 
-	compile('org.springframework.boot:spring-boot-starter-data-jpa')
+    compile('org.springframework.boot:spring-boot-starter-data-jpa')
     compile('org.springframework.boot:spring-boot-starter-web')
     compile('org.springframework.boot:spring-boot-starter-actuator')
+	compile('org.springframework.boot:spring-boot-starter-amqp')
```
Comment (dmssargent) [inline build.gradle]: The whitespacing is off

File: docker-compose.yml [hunk 2]
```diff
@@ -33,6 +33,15 @@ services:
       - mysqlserver
     ports:
       - '8091:8080'
+    environment:
+      - RABBIT_HOST=rabbitmq
+
+
```
Comment (dmssargent) [inline docker-compose.yml]: there is an extra blank line here

**Review by dmssargent — APPROVED** [review]

File: src/main/java/org/aitesting/microservices/authentication/rabbitmq/LoginSuccessListener.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+    @Autowired
+    Producer producer;
+
+    @Override
+    public void onApplicationEvent(AuthenticationSuccessEvent evt) {
+        if (evt.getSource() instanceof UsernamePasswordAuthenticationToken) {
+            User user = (User) evt.getAuthentication().getPrincipal();
+            log.info(String.format("User %s logged in", user.getUsername()));
+            producer.send(String.format("{\"userId\":\"%s\"}", user.getUsername()));
```
Comment (dmssargent) [inline src/main/java/org/aitesting/microservices/authentication/rabbitmq/LoginSuccessListener.java]: You should be using the current user id, not the user name.

**Review by dmssargent — CHANGES_REQUESTED** [review]

**Review by dmssargent — APPROVED** [review]

File: src/main/java/org/aitesting/microservices/authentication/rabbitmq/Producer.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+    @Autowired
+    RabbitTemplate rabbitTemplate;
+
+    @Autowired
+    TopicExchange topicExchange;
+
+    public void send(String message) {
+        log.info(String.format("Sending message %s to %s", message, topicExchange.getName()));
+        this.rabbitTemplate.convertAndSend(topicExchange.getName(), "notification.user.authentication", message);
```
Comment (jacobkissel) [inline src/main/java/org/aitesting/microservices/authentication/rabbitmq/Producer.java]: Routing key should be like this: {bounded-context}.{aggregate-name}.{what-happened-in-past-tense}
So like notification.user.userauthenticated or something maybe

**Review by jacobkissel — DISMISSED** [review]

**Pull request merged**